src/Page.js

```diff
@@ -505,6 +507,42 @@ Page.prototype.removeFrontMatter = function (includedPage) {
   return $.html();
 };
 
+/**
+ * Produces expressive layouts by inserting page data into pre-specified layout
+ * @param pageData a page with its front matter collected
+ */
+Page.prototype.generateExpressiveLayout = function (pageData, fileConfig) {
```

Comment:
It seems like we can't use variables inside `page.md`, so for example if we do:

```
{{ example }}

{{ MAIN_CONTENT_BODY }}

Bottom Content
```

The `{{ example }}` variable won't get rendered.

Reply:
Yes. Was wondering about that. It is okay to support page and global user defined variables. But things can start to get weird when we think about imported/inner variables for this situation.

For example, since the layout and corresponding imports is embedded into all pages, does imported variables come from the layout page or is it part of the actual page? Implementation wise, there should be some additional steps required to prune the import elements from the `page.md` before it is merged with the other content to prevent import error later on.

Reply:
So do you mean that you are worried that in this case, `index.html` will show "This is from b" rather than "This is from c"?

`page.md`
```
<import name from "b.md"/>
{{ name }}

{{ MAIN_CONTENT_BODY }}
```

`index.md`
```
<import name from "c.md"/>
```

`b.md`
```
<variable name="name">This is from b.</variable>
```

`c.md`
```
<variable name="name">This is from c.</variable>
```

Reply:
Hmm, not exactly. The above is resolved. 

It is more of a minor user experience nit about making it clear whether the variable imported from the `page.md` is relative to the `page.md` file while the others are from the actual `index.md`.

Let me enhance the tests to reflect that.